### PR TITLE
fix keyboarding behavior for right and left arrow keys in Tree View

### LINKED ISSUE #5630: fix: left arrow key event on tree item leaf is broken in @microsoft/foundation
# 🐛 Bug Report

When pressing the left arrow key on a tree item leaf, the focus is not moved to the parent item. This is happening on master (c7b30273fe365e998e18e50bb74aa42f38764c05) but not on the [components explorer](https://explore.fast.design/components/fast-tree-view) (presumably because it uses an older version).

## 💻 Repro or Code Sample

- Start the storybook on master for @microsoft/components
- Open the TreeView story
- Expand the tree view
- Select a leaf item
- Press the left arrow key

## 🤔 Expected Behavior

The left arrow on closed and end node should move focus to its parent node ([ref](https://www.w3.org/TR/wai-aria-practices-1.2/#keyboard-interaction-24)).

## 😯 Current Behavior

treeview-left-arrow-key-event

## 🌍 Your Environment

* OS & Device: Debian 11 on PC
* Browser Mozilla FireFox
* Version 96.0.3

### PR DESCRIPTION
**Pull request by chrisdholt**
# Pull Request

## 📖 Description
This PR fixes the left and right arrow behaviors to match those enumerated in the ARIA Authoring Practices document. 

Right Arrow behavior:
> - When focus is on a closed node, opens the node; focus does not move.
> - When focus is on a open node, moves focus to the first child node.
> - When focus is on an end node, does nothing. 

Left Arrow behavior:
> - When focus is on an open node, closes the node.
> - When focus is on a child node that is also either an end node or a closed node, moves focus to its parent node.
> - When focus is on a root node that is also either an end node or a closed node, does nothing.

### 🎫 Issues
fixes #5630 

## 👩‍💻 Reviewer Notes
If you pull down, test and run fast-components, you should see the behavior above for tree items.

## 📑 Test Plan
There are no test for keyboarding behavior currently - we need to add this (ideally w/ playwright)

### General

- [x] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

File: packages/web-components/fast-foundation/src/tree-view/tree-view.ts [hunk 1]
```diff
@@ -147,16 +147,32 @@ export class TreeView extends FoundationElement {
             case keyArrowLeft:
                 if (e.target && this.isFocusableElement(e.target as HTMLElement)) {
                     const item = e.target as HTMLElement;
-                    if (item instanceof TreeItem && item.childItemLength() > 0) {
+
+                    if (
+                        item instanceof TreeItem &&
+                        item.childItemLength() > 0 &&
+                        item.expanded
+                    ) {
```
Comment (chrisdholt) [inline packages/web-components/fast-foundation/src/tree-view/tree-view.ts:155]: Left arrow on an open parent node closes the node.

File: packages/web-components/fast-foundation/src/tree-view/tree-view.ts [hunk 2]
```diff
@@ -147,16 +147,32 @@ export class TreeView extends FoundationElement {
...
                         item.expanded = false;
+                    } else if (
+                        item instanceof TreeItem &&
+                        item.parentElement instanceof TreeItem
+                    ) {
+                        TreeItem.focusItem(item.parentElement);
                     }
```
Comment (chrisdholt) [inline packages/web-components/fast-foundation/src/tree-view/tree-view.ts:162]: Left arrow on either a child node or a closed node moves focus to its parent (if one exists).

File: packages/web-components/fast-foundation/src/tree-view/tree-view.ts [hunk 3]
```diff
@@ -147,16 +147,32 @@ export class TreeView extends FoundationElement {
...
             case keyArrowRight:
                 if (e.target && this.isFocusableElement(e.target as HTMLElement)) {
                     const item = e.target as HTMLElement;
-                    if (item instanceof TreeItem && item.childItemLength() > 0) {
+                    if (
+                        item instanceof TreeItem &&
+                        item.childItemLength() > 0 &&
+                        !item.expanded
+                    ) {
                         item.expanded = true;
```
Comment (chrisdholt) [inline packages/web-components/fast-foundation/src/tree-view/tree-view.ts:173]: Right arrow on a closed node expands the node without moving focus.

File: packages/web-components/fast-foundation/src/tree-view/tree-view.ts [hunk 4]
```diff
@@ -147,16 +147,32 @@ export class TreeView extends FoundationElement {
...
+                    } else if (item instanceof TreeItem && item.childItemLength() > 0) {
+                        this.focusNextNode(1, e.target as TreeItem);
```
Comment (chrisdholt) [inline packages/web-components/fast-foundation/src/tree-view/tree-view.ts:175]: Right arrow on expanded parent node focus its first child node.

**Review by EisenbergEffect — APPROVED** [review]

**Pull request merged**